webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/AccessTokenService.scala

```diff
@@ -21,54 +20,48 @@ object AccessMode extends ExtendedEnumeration {
 
 object AccessResourceType extends ExtendedEnumeration {
   type AccessResourceType = Value
-  val datasource, dataset, tracing, annotation, webknossos, jobExport = Value
+  val dataset, tracing, annotation, webknossos, jobExport = Value
 }
 
 case class UserAccessAnswer(granted: Boolean, msg: Option[String] = None)
 object UserAccessAnswer { implicit val jsonFormat: OFormat[UserAccessAnswer] = Json.format[UserAccessAnswer] }
 
-case class UserAccessRequest(resourceId: DataSourceId, resourceType: AccessResourceType.Value, mode: AccessMode.Value)
+case class UserAccessRequest(resourceId: Option[String], resourceType: AccessResourceType.Value, mode: AccessMode.Value)
 object UserAccessRequest {
   implicit val jsonFormat: OFormat[UserAccessRequest] = Json.format[UserAccessRequest]
 
-  def administrateDataSources: UserAccessRequest =
-    UserAccessRequest(DataSourceId("", ""), AccessResourceType.datasource, AccessMode.administrate)
+  def administrateDatasets: UserAccessRequest =
+    UserAccessRequest(None, AccessResourceType.dataset, AccessMode.administrate)
```

Comment:
My IDE claims "No usage in this Project". Not sure whether thats ok. But why do we have this method anyway. Shouldn't this be a guaranteed reject? Whats the semantic of `None` here? Is ti: I want to administrate all datasets or no dataset at all?

Reply:
Your IDE is mistaken, this is used in DataSourceController.triggerInboxCheckBlocking
This route can be called with an organization id, meaning only the organization directory should be scanned, or without one, meaning everything should be scanned. For the latter option, this access check is used.

I think we might remove this option entirely at some point. Scanning becomes less important and fewer clients call the route, but for now I’d keep it in place, at least for backwards compatibility.